From 4cf1f2c55c8d27052e43da932a5d5444bf1ed1f6 Mon Sep 17 00:00:00 2001
From: Cody Russell <bratsche@gnome.org>
Date: Tue, 01 Jun 2010 15:32:18 +0000
Subject: Bug 608218 - GtkOffscreenWindow causes bad window with GtkEntry

We now exit early from gdk_window_register_dnd() to avoid crashing if the
window type is GDK_WINDOW_OFFSCREEN and does not support dnd operations.
This makes it possible to use any dnd-enabled widgets, such as GtkEntry,
within a GtkOffscreenWindow.
---
diff --git a/gdk/x11/gdkdnd-x11.c b/gdk/x11/gdkdnd-x11.c
index 4b7fd85..c0ad26d 100644
--- a/gdk/x11/gdkdnd-x11.c
+++ b/gdk/x11/gdkdnd-x11.c
@@ -3879,6 +3879,9 @@ gdk_window_register_dnd (GdkWindow      *window)
 
   g_return_if_fail (window != NULL);
 
+  if (gdk_window_get_window_type (window) == GDK_WINDOW_OFFSCREEN)
+    return;
+
   base_precache_atoms (display);
 
   if (g_object_get_data (G_OBJECT (window), "gdk-dnd-registered") != NULL)
--
cgit v0.8.3.1
